src/types.ts

```diff
@@ -403,26 +409,46 @@ export type InitiatorType =
 
 export type NetworkRecordOptions = {
     initiatorTypes?: InitiatorType[]
-    maskRequestFn?: (data: NetworkRequest) => NetworkRequest | undefined
+    maskRequestFn?: (data: CapturedNetworkRequest) => CapturedNetworkRequest | undefined
     recordHeaders?: boolean | { request: boolean; response: boolean }
     recordBody?: boolean | string[] | { request: boolean | string[]; response: boolean | string[] }
     recordInitialRequests?: boolean
+    // whether to record PerformanceEntry events for network requests
+    recordPerformance?: boolean
+    // the PerformanceObserver will only observe these entry types
+    performanceEntryTypeToObserve: string[]
+    // the maximum size of the request/response body to record
+    // NB this will be at most 1MB even if set larger
+    payloadSizeLimitBytes: number
 }
 
-// extending this to match the rrweb NetworkRequest type
-// it is different in that the rrweb type will have initator type, starttime, and endtime
-// as required properties. but we don't want to require them here
-// because we've previously exposed this type as only having `url`
+// deprecated - use CapturedNetworkRequest instead
 export type NetworkRequest = {
     url: string
+}
+
+// In rrweb this is called NetworkRequest but we already exposed that as having only URL
+// we also want to vary from the rrweb NetworkRequest because we want to include
+// all PerformanceEntry properties too.
+// that has 4 required properties
+//     readonly duration: DOMHighResTimeStamp;
+//     readonly entryType: string;
+//     readonly name: string;
+//     readonly startTime: DOMHighResTimeStamp;
+// properties below here are ALPHA, don't rely on them, they may change without notice
+export type CapturedNetworkRequest = Omit<PerformanceEntry, 'toJSON'> & {
```

Comment:
Not important because it covered all important stuff, but in case you are looking for ideas of what to track: https://opentelemetry.io/docs/specs/semconv/http/http-spans/#common-attributes
Eg HTTP version, etc.